js/src/launchAsync.ts

```diff
@@ -225,20 +285,36 @@ export function launchAsync(token: string, subdomain: string, content: Content,
                 }
             }
         };
-        window.addEventListener('message', messageHandler);
+        windowObject.addEventListener('message', messageHandler);
 
         // Reject the promise if the Immersive Reader page fails to load.
-        timeoutId = window.setTimeout((): void => {
+        timeoutId = windowObject.setTimeout((): void => {
             reset();
             isLoading = false;
-            reject({ code: ErrorCode.Timeout, message: `Page failed to load after timeout (${options.timeout} ms)` });
+            if (launchResponseError) {
+                reject(launchResponseError);
+            } else {
+                reject({
+                    code: ErrorCode.Timeout,
+                    message: `Page failed to load after timeout (${options.timeout} ms)`,
```

Comment:
nit: replace  'after' with 'in time and thus timeout'

Reply:
updated